tests/test_infer/test_ops/triton/kernel_utils.py

```diff
@@ -64,10 +64,12 @@ def torch_attn_ref(
 
     assert attn_scores.shape == (bsz, num_heads, q_len, kv_len), "Invalid shape of attention scores"
     if attention_mask is not None:
-        if attention_mask.size() != (bsz, 1, q_len, kv_len):
-            raise ValueError(
-                f"Attention mask should be of size {(bsz, 1, q_len, kv_len)}, but is {attention_mask.size()}"
-            )
+        if q_len == 1:
+            if len(attention_mask.size()) == 4:
+                attention_mask = attention_mask[:, :, -1:, :]
+            else:
+                attention_mask = attention_mask[:, -1:, :]
+
```

Comment:
Wonder why adding this part? Attention masks with unusual forms should be transformed before the `torch_attn_ref`. If alibi has something to do with attention mask, put the logic outside this reference function.

Reply:
After adding alibi_mask, the shape of the last two dimensions of attention_mask will become (max_seq_len, max_seq_len).

Reply:
Can you put the logic outside the function `torch_attn_ref`? If any specific transformations exist for a specific test, put the logic in that test.

Reply:
No, the original assert statement sets num_head to 1 because the masks for all head_dims in the head dimension are consistent and can be automatically expanded. However, in alibi flash attn, the mask for each head's head dim is not consistent, preventing automatic expansion. After adding the alibi mask, the masks for the head dims corresponding to all heads will be generated, eliminating the need for one-dimensional expansion.

Reply:
From your description and your code, I think you could prevent adding the if block in this function. Just put the transformation outside `torch_attn_ref`

Reply:
However, the assert statement mentioned above may need to be deleted.

Reply:
"I think you could **prevent adding the if block in this function**. Just put the transformation outside torch_attn_ref"
--------
Please at least read the whole sentence...